src/world/World.php

```diff
@@ -947,6 +947,16 @@ public function clearCache(bool $force = false) : void{
 		}
 	}
 
+	public function clearBlockCache(int $chunkX, int $chunkZ) : void{
+		$chunk = $this->getChunk($chunkX, $chunkZ);
+		unset($this->blockCache[World::chunkHash($chunkX, $chunkZ)]);
+		if($chunk !== null){
+			foreach($this->getChunkPlayers($chunkX, $chunkZ) as $p){
+				$p->onChunkChanged($chunkX, $chunkZ, $chunk);
+			}
```

Comment:
Since I don't know how to clear the cache of ChunkCache.php directly, I decided to call the onChunkChanged function of the player based on the surrounding code. (If my code reading comprehension is correct).

Reply:
This is already possible to do with plugin api. Previously I wanted add that method just for to avoid doing stuff like [this](https://github.com/CzechPMDevs/BuilderTools/blob/default/src/czechpmdevs/buildertools/editors/object/FillSession.php#L187-L216). Method to clear chunk cache is removed in 4.0.0, that was also reason why I wanted make it for 3.x

Reply:
It's much easier to just use setChunk() instead of that big mess ...